### PR TITLE
316-Improved test coverage

### LINKED ISSUE #316: Improve file transaction logger test coverage
**Detailed Description**
Test coverage for file transactions logging is incomplete.

**Actual Behavior**
Run `RecordFileLoggerFileTest` and witness test coverage is incomplete.
 
**Expected Behavior**
Test coverage should be as complete as possible.

**Environment:**
- Version: 0.3.0

**Additional Context**
Add any other context about the problem here. Attach any logs here, if applicable.

### LINKED ISSUE #317: Create unit tests for address book updates
**Detailed Description**
Unit tests to validate address book updates are missing

**Actual Behavior**
Run unit tests with coverage enabled, witness address book updated aren't tested.

**Expected Behavior**
Address book updates should be tested

**Environment:**
- Version: 0.3.0

**Additional Context**
Add any other context about the problem here. Attach any logs here, if applicable.

### PR DESCRIPTION
**Pull request by gregscullard**
Signed-off-by: Greg Scullard <gregscullard@hedera.com>

**Detailed description**:
Improves test coverage for file transactions and some general record logging
*Files*
File append to system file
Failed file update
Append to address book
Update system file
Update address book
Failed file delete
System Delete
System undelete
Failed System delete
Failed System undelete

*Record Logging*
record file init
record file duplicate
record file completion with and without hashes
record file rollback
Batch size test (through crypto tests)

**Which issue(s) this PR fixes**:
Fixes #316  Fixes #317 

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [X] Tests updated

### CODE REVIEW

File: src/main/java/com/hedera/recordFileLogger/RecordFileLogger.java [hunk 1]
```diff
@@ -104,7 +104,14 @@
         ,CONSENSUS_TIMESTAMP
     	,LIVEHASH
     }
-
+    
+    public static long getFileId() {
```
Comment (steven-sheehy) [inline src/main/java/com/hedera/recordFileLogger/RecordFileLogger.java]: Remove public from these two as they're just for tests

File: src/test/java/com/hedera/recordLogger/RecordFileLoggerFileTest.java [hunk 2]
```diff
@@ -454,6 +511,121 @@ void fileUpdateAllToExistingTest() throws Exception {
...
+                // Additional entity checks
+                ,() -> assertNull(dbFileEntity.getAutoRenewPeriod())
+                ,() -> assertNull(dbFileEntity.getProxyAccountId())
+                ,() -> assertFalse(dbFileEntity.isDeleted())
+         );
+    }
+
+    @Test
+    void fileAppendToAddressBookTest() throws Exception {
+		ConfigLoader.setAddressBookFile(Utility.getResource("addressbook-recordlogger-test").getAbsolutePath());
```
Comment (steven-sheehy) [inline src/test/java/com/hedera/recordLogger/RecordFileLoggerFileTest.java]: I don't think this file should be checked in as it's just a zero byte file. It's also a bad practice to write to files on the classpath. You just need a temp location to write files to, so I recommend using `@TempDir Path tempDir;` to store the addressbook.

File: src/test/java/com/hedera/recordLogger/RecordFileLoggerTest.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+@Sql("classpath:db/scripts/cleanup.sql") // Class manually commits so have to manually cleanup tables
+public class RecordFileLoggerTest extends AbstractRecordFileLoggerTest {
+
+	//TODO: The following are not yet saved to the mirror node database
```
[the remark this answers is not in the document]
Comment (gregscullard) [reply src/test/java/com/hedera/recordLogger/RecordFileLoggerTest.java]: Done

File: src/test/java/com/hedera/recordLogger/RecordFileLoggerTest.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+		assertTrue(RecordFileLogger.start());
+	}
+
+    @AfterEach
+    void after() {
+    	RecordFileLogger.finish();
+    }
+
+    @Test
+    void recordFileInitTest() throws Exception {
```
[the remark this answers is not in the document]
Comment (gregscullard) [reply src/test/java/com/hedera/recordLogger/RecordFileLoggerTest.java]: Done

File: src/main/java/com/hedera/recordFileLogger/RecordFileLogger.java (hunk 1 above)
Comment (gregscullard) [reply src/main/java/com/hedera/recordFileLogger/RecordFileLogger.java]: If I remove `public`, I can't invoke from the test, or am I missing some testing trick here :)

Comment (steven-sheehy) [reply src/main/java/com/hedera/recordFileLogger/RecordFileLogger.java]: Yes, the test package is named wrong. It should be `recordFileLogger`

Comment (gregscullard) [reply src/main/java/com/hedera/recordFileLogger/RecordFileLogger.java]: Got it

Comment (gregscullard) [reply src/main/java/com/hedera/recordFileLogger/RecordFileLogger.java]: Done

File: src/test/java/com/hedera/recordLogger/RecordFileLoggerFileTest.java (hunk 2 above)
Comment (gregscullard) [reply src/test/java/com/hedera/recordLogger/RecordFileLoggerFileTest.java]: Done, although I am not able to reliably test the result of an append on the address book. Added a note to the #58 issue which relates to address book file update
`Note: Testing of fileAppend that results in an address book append fails checking the results of the append due to `addressBookBytes` being a static. The size of the updated file cannot be reliably compared to the transaction payload size.`
The part of the test which checks for matching transaction payload size and address book file contents has been commented out and marked todo with reference to the issue.

File: src/test/java/com/hedera/recordFileLogger/RecordFileLoggerTest.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    void completeFileNoHashes() throws Exception {
+		assertEquals(INIT_RESULT.OK, RecordFileLogger.initFile("TestFile"));
+		RecordFileLogger.completeFile("", "");
+    	final RecordFile recordFile = recordFileRepository.findById(RecordFileLogger.getFileId()).get();
+    	assertNull(recordFile.getFileHash());
+    	assertNull(recordFile.getPreviousHash());
+    }
+    
+    @Test
+    void ompleteFileWithHashes() throws Exception {
```
Comment (steven-sheehy) [inline src/test/java/com/hedera/recordFileLogger/RecordFileLoggerTest.java]: Spelling

File: src/test/java/com/hedera/recordFileLogger/RecordFileLoggerTest.java [hunk 6]
```diff
@@ -0,0 +1,92 @@
+package com.hedera.recordFileLogger;
+
+
+
+import com.hedera.mirror.domain.RecordFile;
```
Comment (steven-sheehy) [inline src/test/java/com/hedera/recordFileLogger/RecordFileLoggerTest.java]: Remove extra blank lines and move import below header

File: src/test/java/com/hedera/recordFileLogger/RecordFileLoggerFileTest.java [hunk 7]
```diff
@@ -845,9 +1038,130 @@ void fileUpdateKeysToNewTest() throws Exception {
...
+                ,() -> assertFalse(dbFileEntity.isDeleted())
+         );
+    }
+
+    @Test
+    void fileUpdateAddressBook() throws Exception {
+
+        Path addressBookPath = Path.of(tempDir.toString(), "addressbook-update.test");
+        FileUtils.touch(addressBookPath.toFile());
+        parserProperties.getMirrorProperties().setAddressBookPath(addressBookPath);
```
Comment (steven-sheehy) [inline src/test/java/com/hedera/recordFileLogger/RecordFileLoggerFileTest.java]: These 3 lines can be simplified to:
```java
parserProperties.getMirrorProperties().setDataPath(tempDir);
Path addressBookPath = parserProperties.getMirrorProperties().getAddressBookPath();
```
I don't think the touch is necessary.

File: src/test/java/com/hedera/recordFileLogger/RecordFileLoggerTest.java (hunk 5 above)
Comment (gregscullard) [reply src/test/java/com/hedera/recordFileLogger/RecordFileLoggerTest.java]: Done

File: src/test/java/com/hedera/recordFileLogger/RecordFileLoggerTest.java (hunk 6 above)
Comment (gregscullard) [reply src/test/java/com/hedera/recordFileLogger/RecordFileLoggerTest.java]: Done

File: src/test/java/com/hedera/recordFileLogger/RecordFileLoggerFileTest.java (hunk 7 above)
Comment (gregscullard) [reply src/test/java/com/hedera/recordFileLogger/RecordFileLoggerFileTest.java]: Done (although I set the dataPath before each test now)

**Pull request merged**